lib/accessly/policy/base.rb

```diff
@@ -54,27 +54,27 @@ def segment_id
         nil
       end
 
-      def grant(action, object = nil)
-        object_id = _get_object_id(object)
-
-        action_id = if object_id.nil?
-          _get_general_action_id!(action)
+      def can?(action, object = nil)
+        if object.nil?
```

Comment:
You shouldn't need to do this check, actually. When you call `send(action)` OR `send(action, object)` OR `send(action, nil)`, all three call `_can_do_action?(action, object)`, which checks whether `object` is nil and calls the appropriate next step.

Given that, you should be able to call `send("#{action}?", object)` regardless of whether `object` is nil.

Reply:
I was getting ArgumentError's without the check

Reply:
Ah, I think I was thinking of a former implementation of this class, which is an implementation which was refactored into what it is now. I guess this check *is* necessary. Sorry about the bad recommendation!